rustls/src/crypto/ring/hmac.rs

```diff
@@ -5,12 +5,9 @@ use alloc::boxed::Box;
 use super::ring_like;
 use crate::crypto;
 
-#[cfg(feature = "tls12")]
 pub(crate) static HMAC_SHA256: Hmac = Hmac(&ring_like::hmac::HMAC_SHA256);
-#[cfg(feature = "tls12")]
 pub(crate) static HMAC_SHA384: Hmac = Hmac(&ring_like::hmac::HMAC_SHA384);
-#[cfg(test)]
-#[allow(dead_code)] // only for TLS1.2 prf test
+#[allow(dead_code)]
```

Comment:
I added a comment.

I also tried to use a `cfg_attr` to only allow the warning for `not(any(test, feature = "aws_lc_rs"))` - that doesn't work nicely because of the case where both `ring` and `aws-lc-rs` are enabled. 

In that situation we still get a warning because the `hmac.rs` file is used by both ring and aws-lc-rs via the `#[path = "../ring/hmac.rs"]` workaround, and in the ring case it ends up unused but in the aws-lc-rs it isn't. Gating by feature alone isn't sufficient and I can't think of another solution for that so an unconditional allow (as was already used) seems best.